Assets/MRTK/Providers/LeapMotion/Editor/ConfigurationChecker/LeapMotionConfigurationChecker.cs

```diff
@@ -0,0 +1,263 @@
+        private static readonly string[] definitions = { "LEAPMOTIONCORE_PRESENT" };
+        private static bool isLeapInProject = false;
+
+        static LeapMotionConfigurationChecker()
+        {
+            // Check if leap core is in the project
+            isLeapInProject = ReconcileLeapMotionDefine();
+
+            if (isLeapInProject)
+            {
+                RemoveTestingFolders();
+                AddAndUpdateAsmDefs();
+                AddLeapEditorAsmDefs();
+            }
+        }
+
+        /// <summary>
+        /// Ensures that the appropriate symbolic constant is defined based on the presence of the Leap Motion Core Assets.
+        /// </summary>
+        /// <returns>True if the define was added, false otherwise.</returns>
+        private static bool ReconcileLeapMotionDefine()
+        {
+            FileInfo[] files = FileUtilities.FindFilesInAssets(FileName);
+
+            if (files.Length > 0)
+            {
+                ScriptUtilities.AppendScriptingDefinitions(BuildTargetGroup.Standalone, definitions);
+                return true;
+            }
+            else
+            {
+                ScriptUtilities.RemoveScriptingDefinitions(BuildTargetGroup.Standalone, definitions);
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// The Leap Core Assets currently contain multiple folders with tests in them.  An issue has been filed in the Unity
+        /// Modules repo: https://github.com/leapmotion/UnityModules/issues/1097.  The issue with the multiple test folders is when an 
+        /// asmdef is placed at the root of the core assets, each folder containing tests needs another separate asmdef.  This method
+        /// is used to avoid adding an additional 8 asmdefs to the project, by removing the folders and files that are tests in the 
+        /// Leap Core Assets.
+        /// </summary>
+        private static void RemoveTestingFolders()
+        {
+            string[] pathsToDelete = new string[]
+            {
+                "LeapMotion/Core/Editor/Tests/",
+                "LeapMotion/Core/Scripts/Algorithms/Editor/Tests/",
+                "LeapMotion/Core/Scripts/DataStructures/Editor/Tests/",
+                "LeapMotion/Core/Scripts/Encoding/Editor/",
+                "LeapMotion/Core/Query/Editor/",
+                "LeapMotion/Core/Scripts/Utils/Editor/BitConverterNonAllocTests.cs",
+                "LeapMotion/Core/Scripts/Utils/Editor/ListAndArrayExtensionTests.cs",
+                "LeapMotion/Core/Scripts/Utils/Editor/TransformUtilTests.cs",
+                "LeapMotion/Core/Scripts/Utils/Editor/UtilsTests.cs",
+                "LeapMotion/Core/Plugins/LeapCSharp/Editor/Tests/",
+                "LeapMotion/Core/Scripts/Query/Editor/"
+            };
+
+            // If one of the leap test directories exists then the rest have not been deleted
+            if (Directory.Exists(Path.Combine(Application.dataPath, pathsToDelete[0])))
```

Comment:
It's interesting that we depend on LeapMotion being copied directly to the application root - we should call this out explicitly (or do the work to support having it in some other arbitrary root -- maybe based on where README_BEFORE_UPDATING is located?)

Note that we've already gotten feedback in the past of people putting MRTK in different locations in the project structure (completely makes sense) - I would expect those people do the same to leap, and I assume we would also support that.

It's also perfectly reasonable here to say "nah, I don't wanna do that" in which case we should just warn or log or do something to tell people that something is wrong.

Reply:
If we expect the GUIDs to stay consistent, those can be used to find the paths regardless of where they are in the assets hierarchy or project structure. Less discoverable what they mean in the code (would need to document what they represent), but removes any dependency on specific paths.

Reply:
Ah, I see we actually do handle it, it's just this particular line that doesn't handle it

Reply:
The pathDifference var was missing in this line but Kurtis's suggestion to use the GUIDs instead of the path seems like a better solution in case leap changes file locations.

Reply:
Investigated using the GUIDs to retrieve paths and discovered that the leap GUIDs are not consistent.  Therefore, cannot reliably be used for path retrieval.

Added the missing pathDifference var.